src/service.js

```diff
@@ -144,12 +144,14 @@ class Service {
       // previous value. This prevents orphaned documents.
       data[this.id] = id;
     }
+    
+    const selector = Object.assign({}, { [this.id]: id }, typeof params === 'object' ? params : undefined);
```

Comment:
I think using `params.query` here would be better since that is what all other adapters are using.